content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md

```diff
@@ -0,0 +1,331 @@
+      PROXY_CONFIG:                  {"envoyAccessLogService":{"address":"afd4a163a65c74af5ad732fcf86b7dff-261027448.us-east-2.elb.amazonaws.com:11800"},"envoyMetricsService":{"address":"afd4a163a65c74af5ad732fcf86b7dff-261027448.us-east-2.elb.amazonaws.com:11800"},"proxyStatsMatcher":{"inclusionRegexps":[".*"]}}
+                                     
+      ISTIO_META_POD_PORTS:          [
+                                         {"containerPort":3550,"protocol":"TCP"}
+                                     ]
+      ISTIO_META_APP_CONTAINERS:     server
+      ISTIO_META_CLUSTER_ID:         Kubernetes
+      ISTIO_META_INTERCEPTION_MODE:  REDIRECT
+      ISTIO_META_WORKLOAD_NAME:      productcatalogservice
+      ISTIO_META_OWNER:              kubernetes://apis/apps/v1/namespaces/boutique/deployments/productcatalogservice
+      ISTIO_META_MESH_ID:            cluster.local
+      TRUST_DOMAIN:                  cluster.local```
+````
+
+It should be noted that, in order to avoid collecting ALS logs for the Skupper deployments, e.g., *skupper-router* & *skupper-service-controller* deployed to the same namespace that are used for multicloud communication and hybrid application deployment, we can add the annotation *sidecar.istio.io/inject: false* to those _Skupper_ deployments. After restart deployments, the _Skupper_ deployments will not inject the *istio-proxy* to their pods. 
+
+#### 3 Monitor Online Boutique application
+
+After follow up all the steps, the Online Boutique application metrics will be showing up from Apache SkyWalking UI, under *Service Mesh* tab:
+
+![](/img/sw-app-svc.png)
+
+The SkyWalking UI *Topology* page will show the Online Boutique application topology map: 
+
+![](/img/sw-app-map.png)
+
+You can check the service *Overview* and *Endpoint* pages per service, e.g., *frontend*:
+
+![](/img/sw-app-overview.png)
+
+![](/img/sw-app-endpoint.png)
+
+You can also check all the services from the *Data Plane* tab under  *Service Mesh* that are observed through _Envoy Metrics Service_ setup:
+
+![](/img/sw-app-svc-dataplane.png)
+
+### Online Boutique application alerting
+
+
+
+Apache SkyWalking provides an alerting mechanism to measure application performance according to a list of pre-defined metrics, e.g., *service\_resp\_time*, *service\_instance\_resp\_time*, and *service\_sla*. It will trigger alerting when some metrics reach its pre-defined thresholds. 
+
+A﻿pache SkyWalking configures the alerting using a collection of alerting rules located in */skywalking/config/alarm-settings.yml* from the SkyWalking OAP pod. You can check the content by running the following command:
+
+
+
+```markdown
+$ kukectl exec pod/skywalking-skywalking-helm-oap-bfb57fbf8-5g7k7 -n skywalking -it -- cat /skywalking/config/alarm-settings.yml 
+```
+You can define new metrics by adding new entry to the file using SkyWalking observability analysis language (OAL), or customize the existing metrics with new thresholds. 
```

Comment:
Add the word "a" prior to "new entry"
Add the word "the" prior to "SkyWalking observability analysis language"